Adjust preliminary test to remove synthetic test data from train set

e13h opened the pull request:
Refactored `example.py` to make it a little easier to follow.

Now, when rebalancing the data, it only rebalances the training set (not the whole data frame). It also uses the exact same train/test split as when fitting unbalanced data, so there is a more uniform comparison.

Closes #7

Commit:
fixed semantic error of rebalancing all data and training on all synthetic data
also refactored example.py for simplicity

e13h committed:
Clean up imports

Commit:
fix bug in rebalance.py

Commit:
Merge branch 'example-rebalance-train-set-only' of https://github.com/byu-dml/d3m-profiler into example-rebalance-train-set-only

bjschoenfeld commented on README.md:
````diff
@@ -21,3 +21,12 @@ Run models with `python3 evaluate_models.py [<number of cores to use>]`
+## Creating a `conda` environment
+To create a `conda` virtual environment for running `example.py`, run the following commands:
+```
+conda create --name example cudatoolkit=9.0 tensorflow tensorflow-gpu cycler joblib kiwisolver matplotlib numpy pandas pyparsing python-dateutil pytz scikit-learn scipy six
+
+conda activate example
+pip install imbalanced-learn==0.6.2 sent2vec-prebuilt==0.0.3
````
Can conda use the dependencies listed in `requirements.txt`?

e13h replied:
I ran into some errors because there are dependencies in our `requirements.txt` that have a version of `0.0.0` for some reason. `pip` handles this as "oh, that dependency is met" while `conda` says "oh no, can't find that version"
https://github.com/byu-dml/d3m-profiler/blob/9a3bc45061267091b0109f2159648785e370a18b/requirements.txt#L8
https://github.com/byu-dml/d3m-profiler/blob/9a3bc45061267091b0109f2159648785e370a18b/requirements.txt#L16

bjschoenfeld replied:
Ah yes, I have seen this issue before. Both of those dependencies should be removed from `requirements.txt`. Can you remove them and then update the readme to use the requirements.txt?

bjschoenfeld commented on d3m_profiler/rebalance.py:
```diff
@@ -137,15 +137,18 @@ def _construct_rebalanced_df(X_resampled: pd.DataFrame, Y_resampled: pd.DataFram
     The rebalanced DataFrame 
 """
-def rebalance_SMOTE(df: pd.DataFrame, type_column: str, method: str, model_weights_path: str) -> pd.DataFrame:
+def embed_and_rebalance_SMOTE(df: pd.DataFrame, type_column: str, method: str, model_weights_path: str) -> pd.DataFrame:
     embedded_df = embed(df, type_column, model_weights_path)
 
+    return rebalance_SMOTE(embedded_df, type_column, method)
+    
+
+def rebalance_SMOTE(embedded_df: pd.DataFrame, type_column: str, method: str) -> pd.DataFrame:
```
Perhaps we should decouple the embedding and the rebalancing functions.

bjschoenfeld commented on example.py:
```diff
@@ -31,74 +26,60 @@
-open_d3m_file = 'data/open_d3m_data.csv'
-closed_d3m_file = 'data/closed_d3m_data.csv'
+open_d3m_file = '~/data/open_d3m_unembed_data.csv'
+closed_d3m_file = '~/data/closed_d3m_unembed_data.csv'
```
can we add the data to this repo? maybe a compressed version.

bjschoenfeld replied:
can we also rename `open` to `public` and `closed` to `private`

e13h replied:
@bjschoenfeld is it okay if the metadata for the private database (`closed_d3m_unembed_data.csv` above) is hosted on our repo? It's not _the_ data, but it includes all the private dataset names and descriptions.

e13h replied:
@bjschoenfeld I uploaded a compressed version of the public metadata `.csv` file. Let me know if I have the go-ahead to do the same for the private metadata `.csv` file too.

bjschoenfeld replied:
maybe to be safe, let's not publish the private one.

bjschoenfeld commented on example.py:
```diff
@@ -31,74 +26,60 @@
+
+    return {'data_collection': data_collection, 'classifier': model_class.__name__, 'balanced': isBalanced, 'accuracy_score': accuracy, 'f1_score_micro': f1_micro, 'f1_score_macro': f1_macro, 'f1_score_weighted': f1_weighted}
+
+def split_X_and_y(data: pd.DataFrame):
+    X = data.drop(['datasetName', type_column], axis=1)
+    y = data[type_column]
+    return X, y
```
the name `split` has a connotation of train/test split. is there another name we could use? `extract`? `separate`?

e13h committed:
Simplify rebalance and embed functions
Since y isn't being embedded, it isn't passed to embed() anymore.
Instead of passing a single dataframe to rebalance() and relying on hard-coded data extraction, now you simply pass in X and y that you want rebalanced

e13h committed:
Refactored example.py to simplify testing structure
Include public metadata .csv in repo
remove unused (and clunky) embed_and_reblance_SMOTE() function from rebalance.py

e13h committed:
automatically create the private metadata file if it doesn't already exist

Commit:
remove dependencies set with version 0.0
update readme accordingly to create virtual environments with requirements.txt

e13h commented:
Closed because of staleness
